node.go

```diff
@@ -27,27 +27,24 @@ func NewNode(expression exp.Expression) *Node {
 	}
 }
 
-/*
-	if t.Expression == nil {
-		return nil
-	}
-*/
 func (t Node) Evaluate(ctx exp.Context, list *[]string) error {
 	result, err := toBool(t.Expression.Evaluate(ctx))
 	if err != nil {
 		return errors.Wrapf(err, "failed to evaluate expression %v", t.Expression)
 	}
 
-	if result {
-		*list = append(*list, t.Payload...)
-	}
+	// if result {
+	// 	*list = append(*list, t.Payload...)
+	// }
```

Comment:
I don't think we should be commenting out this.  As we walk down the tree we should be adding the values to the list.